### PR TITLE
Add HTTPS/Basic auth support

### PR DESCRIPTION
**Pull request by roidelapluie**
Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

**Review by beorn7 — APPROVED** [review]
Just documentation nits. 

Very awesome. Please merge once you have addressed my nits one way or the other.

**Comment by roidelapluie** [comment]
I have added a sentence to clarify that everything is protected by the TLS/Basic auth, including scrape and UI, and not only pushing metrics.

File: README.md [hunk 1]
```diff
@@ -593,6 +593,18 @@ metrics. Those pushes are still counted as
 the `rate` of this metric, but you have to inspect the logs to identify the
 offending pusher.
 
+## TLS and basic authentication
+
+The Pushgateway supports TLS and basic authentication. This enables better
+control of the API's.
```
Comment (beorn7) [inline README.md:599]: If I may continue my nitpicking: API's → APIs.

However, the `/metrics` endpoint is not really an API, is it? How about "This enables better control of the various HTTP endpoints."

File: README.md [hunk 2]
```diff
@@ -593,6 +593,18 @@ metrics. Those pushes are still counted as
...
+
+To use TLS and/or basic authentication, you need to pass a configuration file
+using the `--web.config.file` parameter. The format of the file is described
+[in the exporter-toolkit repository](https://github.com/prometheus/exporter-toolkit/blob/master/docs/web-configuration.md).
+
+Note that the TLS and basic authentication settings affects the API's to
+push metrics, the /metrics endpoint used by Prometheus and the web UI.
```
Comment (beorn7) [inline README.md:606]: Various nits here. My suggestion:

Note that the TLS and basic authentication settings affect all HTTP endpoints: `/metrics` for scraping, the API to push metrics via `/metrics/...`, the admin API via `/api/...`, and the web UI.

**Review by beorn7 — CHANGES_REQUESTED** [review]
Thanks for the doc tweaks. Which inspired me to tweak them even more. (o:

**Comment by roidelapluie** [comment]
Awesome, applied!

**Review by beorn7 — APPROVED** [review]

**Pull request merged**